py-polars/polars/internals/frame.py

```diff
@@ -98,10 +100,22 @@
 else:
     from typing_extensions import Literal
 
+if sys.version_info >= (3, 10):
+    from typing import TypeAlias
+else:
+    from typing_extensions import TypeAlias
+
 # A type variable used to refer to a polars.DataFrame or any subclass of it.
 # Used to annotate DataFrame methods which returns the same type as self.
 DF = TypeVar("DF", bound="DataFrame")
 
+if TYPE_CHECKING:
+    # X and Y are used to annotate DataFrame.__getitem__()
+    # X indexes into the vertical axis and Y indexes into the horizontal axis
+    # NOTE: making these strings is necessary for Python <3.10
+    X: TypeAlias = "slice | range | list[int] | list[bool] | pli.Series"
+    Y: TypeAlias = "slice | range | list[int] | list[bool] | list[str] | pli.Series"
```

Comment:
Can we give these two more descriptive names, such as `RowIterableIndex` and `ColIterableIndex`? (although Iterable is not exactly the right word, maybe you can think of a better one.

I find type aliases in general not great, as it is another layer as a user to peel back on, but I can see the need for it here given the vast number of input types supported.

Reply:
`MultiRowIndexer` and `MultiColIndexer`? 

Personally, I prefer `X` and `Y`. I consider these two variables defined **purely** for the benefit of the linter, so short names lead to more succinct code. In fact, PyCharm automatically expands these out, so a user is never left wondering "what is X?"

<img width="1080" alt="Screen Shot 2022-08-03 at 5 08 26 PM" src="https://user-images.githubusercontent.com/13473688/182712127-e1842981-1577-4528-a093-b6045d6c6081.png">

But this is an example of the classic debate of "are type hints meant for humans, or for analyzers?" I am happy to make the switch.

Reply:
Multi is a better word indeed. For the users, where the IDE/editor may not auto expand (such as VS Code, at least it doesn't for me), I strongly prefer the more descriptive type alias, especially as the definition and usage are 1700 lines of code apart. For contributors, it is like naming (global) variables, where we also prefer descriptive names over saying "variables are for analyzers/interpreters". The analyzer doesn't care about a few extra characters.